scipy/stats/tests/test_morestats.py

```diff
@@ -1987,6 +1987,88 @@ def test_circfuncs_unit8(self):
         assert_allclose(stats.circstd(x, high=180), 20.91551378, rtol=1e-7)
 
 
+class TestQuartileCoeffOfDispersion(object):
```

Comment:
Tests are carefully explained; thanks!

I think they're sufficient. For instance, I don't think there's a need to test exotic nd-inputs because you've already shown correct behavior for 2d and the code is straightforward enough to see that it should work as long as `np.quantile` does what it says it does:
> If multiple quantiles are given, first axis of the result corresponds to the quantiles. The other axes are the axes that remain after the reduction of _a._